addons/html_editor/static/src/others/embedded_components/plugins/excalidraw_plugin/excalidraw_dialog/excalidraw_dialog.js

```diff
@@ -20,7 +20,29 @@ export class ExcalidrawDialog extends Component {
         super.setup();
         this.state = useState({});
         this.inputRef = useAutofocus({ refName: "urlInput" });
+        this.insertButtonRef = useRef("insertButton");
         useExternalListener(window, "keydown", this.onKeyDown.bind(this));
+        useEffect(
+            (inputEl) => {
+                if (inputEl) {
+                    const onInput = () => {
+                        const value = !this.inputRef.el.value || this.inputRef.el.value.trim() === "";
+                        this.insertButtonRef.el.toggleAttribute("disabled", value);
+                    };
+
+                    inputEl.addEventListener("input", onInput);
+                    return () => {
+                        inputEl.removeEventListener("input", onInput);
+                    }
+                }
+            },
+            () => [this.inputRef.el]
+        );
+        onMounted(() => {
+            if (this.insertButtonRef.el) {
+                this.insertButtonRef.el.setAttribute("disabled", true);
+            }
+        });
```

Comment:
Do you really need to add all this? Why not use a value in the state that is updated based on the inputRef value?

Reply:
Hey @thjo-odoo, 

Yes, it is needed. we want to disable the insert btn when URL input is empty.

Reply:
That's not the question, why does it have to be so complicated? 
Why not use the already present onKeyDown method to handle the behavior via a state value read inside the template?

Reply:
Hey @thjo-odoo, 

I have tuned the code based on the remarks. It was malfunctioning on `copy-paste' operation and every subsequent key typing the button was not disabling as expected behavior on onKeyDown method. so I have achieved the behavior via
checkInput method. it is working great. 

Thank you for the suggestion as the code has been reduced and cleaned.

Please have a look.

Thanks